extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/ColorThingHandler.java

```diff
@@ -0,0 +1,300 @@
+            if (turnOnValue.size() == 3) {
+                this.turnOnValue = turnOnValue;
+                logger.trace("set turnonvalue to {} in {}", turnOnValue, this.thing.getUID());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "turn-on value malformed");
+                dmxHandlerStatus = ThingStatusDetail.CONFIGURATION_ERROR;
+                return;
+            }
+        }
+        this.turnOnValue.setFadeTime(fadeTime);
+
+        if (configuration.get(CONFIG_DIMMER_TURNOFFVALUE) != null) {
+            String turnOffValueString = String.valueOf(fadeTime) + ":"
+                    + ((String) configuration.get(CONFIG_DIMMER_TURNOFFVALUE)) + ":-1";
+            ValueSet turnOffValue = ValueSet.fromString(turnOffValueString);
+            if (turnOffValue.size() == 3) {
+                this.turnOffValue = turnOffValue;
+                logger.trace("set turnoffvalue to {} in {}", turnOffValue, this.thing.getUID());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "turn-off value malformed");
+                dmxHandlerStatus = ThingStatusDetail.CONFIGURATION_ERROR;
+                return;
+            }
+        }
+        this.turnOffValue.setFadeTime(fadeTime);
+
+        if (bridge.getStatus().equals(ThingStatus.ONLINE)) {
+            updateStatus(ThingStatus.ONLINE);
+            dmxHandlerStatus = ThingStatusDetail.NONE;
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (channels.size() != 0) {
+            channels.get(0).removeListener(new ChannelUID(this.thing.getUID(), CHANNEL_BRIGHTNESS_R));
+            channels.get(1).removeListener(new ChannelUID(this.thing.getUID(), CHANNEL_BRIGHTNESS_G));
+            channels.get(2).removeListener(new ChannelUID(this.thing.getUID(), CHANNEL_BRIGHTNESS_B));
+        }
+        channels.clear();
+        currentValues.clear();
+        currentColor = new HSBType();
+    }
+
+    @Override
+    public void updateChannelValue(ChannelUID channelUID, int value) {
+        updateState(channelUID, Util.toPercentValue(value));
+        switch (channelUID.getId()) {
+            case CHANNEL_BRIGHTNESS_R:
+                currentValues.set(0, value);
+                break;
+            case CHANNEL_BRIGHTNESS_G:
+                currentValues.set(1, value);
+                break;
+            case CHANNEL_BRIGHTNESS_B:
+                currentValues.set(2, value);
+                break;
+            default:
+                logger.debug("don't know how to handle {} in RGB type", channelUID.getId());
+                return;
+        }
+        currentColor = HSBType.fromRGB(currentValues.get(0), currentValues.get(1), currentValues.get(2));
+        updateState(new ChannelUID(this.thing.getUID(), CHANNEL_COLOR), currentColor);
```

Comment:
The `COLOR`  channel is high level channel for convient use of the thing. Its value is calculated from the lower level channels (i.e. the R,G,B components). 
The feedback is only provided on a physical channel (i.e.  R,G,B), but the user wants to see the change in the `COLOR`, so on receiving one feedback, both need to be updated. This could only be avoided if the low-level channels are not exposed to the user, but IMO this would impose a limit on functionality for the sake of "nicer" code. 
The same is true for the `TunableWhiteThingHandler` below: the physical channels are cold white and warm white, the total brightness ond the color temperature is only calculated from those.